lesson01/main.tf

```diff
@@ -0,0 +1,73 @@
+resource "aws_default_vpc" "default" {
+  tags = {
+    Name = "Default VPC"
+  }
+}
+
+resource "aws_security_group" "web" {
+  name        = "web"
+  description = "Allow web traffic"
+  vpc_id      = "${aws_default_vpc.default.id}"
+
+  ingress {
+    from_port   = 80
+    to_port     = 80
+    protocol    = "TCP"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags = {
+    Terraform  = true
+    Service    = "web"
+    Customer   = "Stark Industries"
+    Owner      = "Wizeline"
+    Maintainer = "devops@wizeline.com"
+  }
+}
+
+resource "aws_instance" "web" {
+  ami           = "${data.aws_ami.amazon_linux.id}"
+  instance_type = "t2.micro"
+
+  provisioner "local-exec" {
```

Comment:
I would prefer if we show them how to use user_data and a template file instead of local provisioner 
data "template_file" "deploy_sh" {
  template = "${file("${path.module}/userdata.sh")}"
  vars {
    docker_tag      = "${var.appversion}"
   docker_image      = "${var.docker_image}"
    HOSTNAME        = "${var.metadata["appname"]}-${var.env}-ec2-${var.appversion}"
  }
}

Reply:
we can have both, as user_data is aws only afaik and I'm not sure if that applies to other cloud providers so would be good to know both ways

Reply:
The local-exec provisioner is used to wait until the docker container is running and the instance is ready to accept connections. 

I will work on creating the template file for the user data.